internal/explorer/helpers.go

```diff
@@ -175,3 +178,148 @@ func (a *App) handleRequestsQueryParams(r *http.Request) (*http.Request, error)
 	ctx = context.WithValue(ctx, isGatewayKey{}, isGateway)
 	return r.WithContext(ctx), nil
 }
+
+// fetchNodeData is a helper method that fetches nodes data over rmb
+// returns the node capacity, hypervisor and dmi
+func (a *App) fetchNodeData(nodeID string) (NodeInfo, error) {
+	twinID, err := a.getNodeTwinID(nodeID)
+	if err != nil {
+		return NodeInfo{}, ErrNodeNotFound
```

Comment:
If graphql is down, this will return ErrNodeNotFound. the `getNodeTwinID` knows the exact error, it can be the one to return `ErrNodeNotFound` in case the list of returned node length is zero and this method can just propagate the returned error.

Reply:
yes thanks

Reply:
done